secboot/secboot_tpm.go

```diff
@@ -21,19 +21,50 @@
 package secboot
 
 import (
+	"errors"
 	"fmt"
+	"os"
 
 	sb "github.com/snapcore/secboot"
 
 	"github.com/snapcore/snapd/logger"
 )
 
+var secbootConnectToDefaultTPM = sb.ConnectToDefaultTPM
+
 func CheckKeySealingSupported() error {
-	logger.Noticef("checking TPM device availability...")
-	tconn, err := sb.ConnectToDefaultTPM()
+	logger.Noticef("checking if secure boot is enabled...")
+	if err := checkSecureBootEnabled(); err != nil {
+		return err
+	}
+	logger.Noticef("secure boot is enabled")
+
+	logger.Noticef("checking if TPM device is available...")
+	tconn, err := secbootConnectToDefaultTPM()
 	if err != nil {
 		return fmt.Errorf("cannot connect to TPM device: %v", err)
 	}
 	logger.Noticef("TPM device detected")
 	return tconn.Close()
 }
+
+var efivarsSecureBootFile = "/sys/firmware/efi/efivars/SecureBoot-8be4df61-93ca-11d2-aa0d-00e098032b8c"
+
+func checkSecureBootEnabled() error {
+	f, err := os.Open(efivarsSecureBootFile)
+	if err != nil {
+		return fmt.Errorf("cannot open secure boot file: %v", err)
+	}
+	defer f.Close()
+
+	buf := make([]uint8, 5)
```

Comment:
I was wondering about the format here, so went on digging. The UEFI spec states:

> The SecureBoot variable is an 8-bit unsigned integer that defines whether the platform firmware is operating with Secure Boot enabled. A value of 1 indicates that platform firmware performs driver and boot application signature verification as specified in UEFI Image Validation during the current boot.

However, the code below reads `buf[4]`. The efivars fs is somewhat undocumented, so had to go the source and found this: https://elixir.bootlin.com/linux/latest/source/fs/efivarfs/file.c#L91The actual content read is `<u32 attributes><n bytes of data>`.

I think we should add a comment about the format of the data read from `/sys/..`

Reply:
@bboozzoo do you have something like:
```
For details of the format see https://elixir.bootlin.com/linux/v5.6.3/source/fs/efivarfs/file.c#L91
```
in mind?

Reply:
this will do:
```
// the format of data read from efivars is: <u32 attribute>:<n bytes of data>
// for details see: https://elixir.bootlin.com/linux/v5.6.3/source/fs/efivarfs/file.c
```

Reply:
not a blocker for this, but we probably want some helpers to read efi vars, I think @anonymouse64 also needs that (though utf16 not, numbers),  maybe we want them in bootloader/efi ?

Reply:
yes I have those helpers to read generic binary data from the efi vars in https://github.com/snapcore/snapd/pull/8424/files#diff-3f706a2511358274ad3afdefd9226c19

Reply:
Added the comment. Also will refactor to use `ReadEfiVar()` from #8424 after that PR lands.